specs/Standards/Tokens/NonFungibleToken.md

```diff
@@ -0,0 +1,167 @@
+- Alice's account is `alice`
+- Jeraldo's account is `jerry`
+- The Escrow contract is `escrow`
+- The NFT contract has been initialized with a non-zero token supply
+- There exists a Corgi token with the ID of `3` and a Sausage token with the ID of `5`
+- The Escrow contract manages how the transfer is facilitated and guarantees requirements are met for transfer
+
+#### **High-level**
+
+Both Alice and Jerry will issue asynchronous transactions to their respective contracts, `corgi` and `sausage` to grant access to the escrow to trade tokens on their behalf. `escrow` will call the `sasuage` token contract asynchrounously to transfer the Sausage token to `escrow`. After, `escrow` will also call the `corgi` contract to asynchornously transfer the Corgi token to `escrow`. Then, `escrow` will conduct a transfer to both parties.
+
+- If both of the `transfer` calls succeed, then Alice will now own one Sausage token and Jerry will own one Corgi token.
+- If one or both of the final `transfer` calls fail, then nothing will happen and `escrow` should attempt reissuing the failed transaction.
+
+#### **Technical calls**
+
+1. `alice` makes an async call to `corgi::grant_access({"escrow_account_id":"escrow"})`
+2. `jerry`  makes an async call to `sausage::grant_access({"escrow_account_id":"escrow"})`
+3. `escrow` calls `sausage::transfer({"new_owner_id:"escrow", "token_id": 5})`
+    - Recommondation: attach callback `escrow::on_transfer({"owner_id":"jerry", "token_contract":"sausage", "token_id": 5})`
+4. `escrow` calls `corgi::transfer({"new_owner_id:"escrow", "token_id": 3})`
+    - Recommendation: attach callback `escrow::on_transfer({"owner_id":"alice", "token_contract":"corgi", "token_id": 3})`
+5. In one Promise:
+    1. `escrow` calls `corgi::transfer({"new_owner_id:"jerry", "token_id": 3})`
+        - attaches callback `escrow::on_transfer({"owner_id":"alice", "token_contract:"corgi", "token_id": 3})`
+    2. `escrow` calls `sausage::transfer({"new_owner_id:"escrow", "token_id": 5})`
+        - attaches callback `escrow::on_transfer({"owner_id":"jerry", "token_contract":"corgi", "token_id": 3})`
+
+
+## Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+## Template for smart contract in AssemblyScript
+
+At time of writing, this standard is established with several constraints found in AssemblyScript. The first is that interfaces are not an implemented feature of AssemblyScript, and the second is that classes are not exported in the conversion from AssemblyScript to WASM. This means that the entire contract could be implemented as a class, which might be better for code organization, but it would be deceiving in function.
+
+```TypeScript
+
+/******************/
+/* CHANGE METHODS */
+/******************/
+
+// Grant access to the given `escrow_account_id` for all tokens that account has.
+// Requirements:
+// * The caller of the function (`predecessor_id`) should have access to the tokens.
```

Comment:
That's a little confusing in terms or requirements and part for `for all tokens that account has.` Maybe better to restructure it, to remove requirements and explain that the `predecessor_id` gives access to `escrow_account_id` for all tokens